src/Sylius/Bundle/ResourceBundle/ExpressionLanguage/NotNullResultExpressionLanguage.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\ExpressionLanguage;
+
+use Symfony\Component\DependencyInjection\ExpressionLanguage as BaseExpressionLanguage;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+final class NotNullResultExpressionLanguage extends BaseExpressionLanguage
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws NotFoundHttpException if expression returns null value
+     */
+    public function evaluate($expression, $values = array())
+    {
+        $result = parent::evaluate($expression, $values);
+        if (null === $result) {
```

Comment:
IMHO it should be a function `throwNotFoundExceptionOnNull(value)` which returns value if `value != null` or throws an exception if `value == null`, wdyt?